controllers/cluster_controller.go

```diff
@@ -209,7 +210,7 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	if cluster.Spec.ControlPlaneRef != nil {
 		obj, err := external.Get(ctx, r.Client, cluster.Spec.ControlPlaneRef, cluster.Namespace)
 		switch {
-		case apierrors.IsNotFound(err):
+		case apierrors.IsNotFound(errors.Cause(err)):
```

Comment:
This is a bug that has cropped up on more than one occasion, I'm wondering if we should have a wrapper for this in our errors package (aliased capierrors here) that handles extracting the causing error?

Reply:
Yeah I was thinking the same the other day, we can go the alias route and have one `errors` package that aliases all the things

Reply:
I'm going to open an issue around this. I think we could fix it an alternative way. I don't think this should be a blocker for this PR.

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/2138